echo_server.py

```diff
@@ -19,8 +71,15 @@
                 msg_part = conn.recv(buffsize)
                 msg += msg_part
                 if len(msg_part) < buffsize:
+                    # print 'msg:\n{}'.format(msg)
                     done = True
-                    conn.sendall(msg)
+                    first_line = parse_request2(msg)
+                    try:
+                        out = check_request(first_line)
+                    except (Error405, Error505) as e:
+                        out = str(e)
```

Comment:
The try/except here is correct, but parse_request2 and check_request should be combined into a single function call on line 78 and then 79 should be where out is bound to the result of calling `response_ok`.